packages/model-viewer/src/features/scene-graph.ts

```diff
@@ -139,21 +138,19 @@ export const SceneGraphMixin = <T extends Constructor<ModelViewerElementBase>>(
           return;
         }
 
-        const updatedMaterialsPromise =
-            threeGLTF.correlatedSceneGraph.loadVariant(
-                variantName!, this[$getOnUpdateMethod]);
+        const updatedMaterials =
+            await threeGLTF.correlatedSceneGraph.loadVariant(variantName!);
         const {gltf, gltfElementMap} = threeGLTF.correlatedSceneGraph;
 
-        updatedMaterialsPromise.then(updatedMaterials => {
```

Comment:
Switching from `.then` to async/await since it's a bit easier to read and it matches our style.